pkg/request/pending.go

```diff
@@ -37,6 +41,9 @@ type PendingStore struct {
 	processed             sync.Map
 	currentBucket         atomic.Value
 	buckets               []*bucket
+	stopped               uint32
+	closed                uint32
+	bucketsLock           sync.RWMutex
```

Comment:
the reason for the addition of the lock was that we can now restart it ?

Reply:
I think it was mostly so we can call `GetAllRequests`